src/StackManager.js

```diff
@@ -72,21 +76,26 @@ export default class StackManager {
     const file = dockerComposePath(this.getProjectPath());
     const command = clean(args).join(' ');
 
-    return exec(`docker-compose --file ${file} ${command}`);
+    return exec(`docker-compose -f ${file} -p ${this.getProjectName()} ${command}`);
   }
   spawnDockerCompose(...args) {
     const file = dockerComposePath(this.getProjectPath());
-    const child = spawn('docker-compose', clean(['--file', file, ...args]), { stdio: 'inherit' });
+    const child = spawn('docker-compose', clean([
+      '-f', file,
+      '-p', this.getProjectName(),
+      ...args,
+    ]), { stdio: 'inherit' });
 
     // cancel default SIGINT behaviour
     process.on('SIGINT', () => {});
 
     return child;
   }
-  async listContainers(serviceName = null) {
-    const stdout = await this.execDockerCompose('ps -q', serviceName);
+  async listRunningContainers(serviceName) {
+    const label = `${this.getProjectName()}.${serviceName}`;
+    const stdout = await this.execDocker('ps -q', '--filter', `label=stacker=${label}`, '--filter', 'status=running');
```

Comment:
Also, when the arguments ar inside an array we can arrange them nicely on multiple lines.